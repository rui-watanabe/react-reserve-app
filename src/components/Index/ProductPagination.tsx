import React from 'react';
import { useRouter } from 'next/router';
import { Container, Pagination } from 'semantic-ui-react';
import { ProductPageNationProps } from '../Types/IndexType';

function ProductPagination({
  totalPages,
}: ProductPageNationProps): JSX.Element {
  const router = useRouter();
  return (
    <Container textAlign="center" style={{ margin: '2em' }}>
      {console.log(router)}
      <Pagination
        defaultActivePage={router.query.page ? Number(router.query.page) : 1}
        totalPages={totalPages}
        onPageChange={(event, data) => {
          data.activePage === 1
            ? router.push('/')
            : router.push(`/?page=${data.activePage}`);
        }}
      />
    </Container>
  );
}

export default ProductPagination;
