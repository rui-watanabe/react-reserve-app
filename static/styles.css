:root {
  --bodyBg: #f0f4f8;
  --menuBg: #008080;
  --progressBar: #54d1db;
}

body {
  background: var(--bodyBg);
}

#menu {
  background: var(--menuBg);
}

/* Overrides default styled accordion styles (too much bottom padding) */
.ui.styled.accordion .content {
  margin: 0;
  padding: 0.5em 1em 0.75em !important;
}

.ui.card > .image:not(.ui) > img, .ui.cards > .card > .image:not(.ui) > img  {
  height: 212px;
}
