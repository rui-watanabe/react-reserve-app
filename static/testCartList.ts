import { CartProductType } from '../src/pagesTypes/CartType';
import { testProductList } from './testProductList';

export const testCartList: CartProductType = [
  {
    quantity: 1,
    product: testProductList[0],
  },
  {
    quantity: 2,
    product: testProductList[1],
  },
];
