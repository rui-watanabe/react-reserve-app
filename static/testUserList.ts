import { ObjectId } from 'mongodb';
import { UserModelType } from '../models/Types/UserType';

export const testUserList: UserModelType[] = [
  {
    _id: new ObjectId('000000000000000000000001'),
    createdAt: new Date('2020-9-22'),
    updatedAt: new Date('2020-9-22'),
    name: 'test',
    email: 'test@gmail.com',
    password: '11111111',
    role: 'root',
  },
  {
    _id: new ObjectId('000000000000000000000002'),
    createdAt: new Date('2020-9-22'),
    updatedAt: new Date('2020-9-22'),
    name: 'test2',
    email: 'test2@gmail.com',
    password: '22222222',
    role: 'admin',
  },
  {
    _id: new ObjectId('000000000000000000000003'),
    createdAt: new Date('2020-9-22'),
    updatedAt: new Date('2020-9-22'),
    name: 'test',
    email: 'test@gmail.com',
    password: '33333333',
    role: 'user',
  },
];
