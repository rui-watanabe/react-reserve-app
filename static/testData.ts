import { ProductModelType } from '../models/Types/ProductType';
import { UserModelType } from '../models/Types/UserType';
import { CartProductType } from '../src/pagesTypes/CartType';
import { testProductList } from './testProductList';
import { testUserList } from './testUserList';

export const testProduct: ProductModelType = testProductList[0];

export const testTotalPages = Math.floor(testProductList.length / 9);

export const testUser: UserModelType = testUserList[2]; // 'user' role user

export const testCartProductList: CartProductType = [
  { product: { ...testProduct }, quantity: 1 },
  { product: { ...testProductList[1] }, quantity: 2 },
  { product: { ...testProductList[2] }, quantity: 3 },
];
