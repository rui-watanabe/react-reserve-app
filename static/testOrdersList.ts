import { ObjectId, ObjectID } from 'mongodb';
import { OrderModelType } from '../models/Types/OrderType';
import { testProductList } from './testProductList';
import { testUserList } from './testUserList';

export const testOrdersList: OrderModelType[] = [
  {
    _id: new ObjectId('200000000000000000000001'),
    createdAt: new Date(2020, 4, 1, 12, 30, 45),
    updatedAt: new Date(2020, 4, 1, 12, 30, 45),
    products: [
      {
        _id: new ObjectID('100000000000000000000001'),
        quantity: 1,
        product: testProductList[0],
      },
      {
        _id: new ObjectID('100000000000000000000002'),
        quantity: 2,
        product: testProductList[1],
      },
    ],
    user: testUserList[2]._id, // 'user' role user
    email: testUserList[2].email,
    total: testProductList[0].price + testProductList[1].price * 2,
  },
  {
    _id: new ObjectID('200000000000000000000002'),
    createdAt: new Date(2020, 4, 1, 12, 30, 45),
    updatedAt: new Date(2020, 4, 1, 12, 30, 45),
    products: [
      {
        _id: new ObjectID('100000000000000000000003'),
        quantity: 1,
        product: testProductList[2],
      },
      {
        _id: new ObjectID('100000000000000000000004'),
        quantity: 2,
        product: testProductList[3],
      },
    ],
    user: testUserList[2]._id, // 'user' role user
    email: testUserList[2].email,
    total: testProductList[2].price + testProductList[3].price * 2,
  },
  {
    _id: new ObjectID('200000000000000000000003'),
    createdAt: new Date(2020, 4, 1, 12, 30, 45),
    updatedAt: new Date(2020, 4, 1, 12, 30, 45),
    products: [
      {
        _id: new ObjectID('100000000000000000000005'),
        quantity: 1,
        product: testProductList[4],
      },
      {
        _id: new ObjectID('100000000000000000000006'),
        quantity: 2,
        product: testProductList[5],
      },
    ],
    user: testUserList[2]._id, // 'user' role user
    email: testUserList[2].email,
    total: testProductList[4].price + testProductList[5].price * 2,
  },
];
