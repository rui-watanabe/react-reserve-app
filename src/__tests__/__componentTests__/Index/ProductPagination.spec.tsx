import React from 'react';
import renderer from 'react-test-renderer';
import { testProductList } from '../../../../static/testProductList';
import ProductPagination from '../../../components/Index/ProductPagination';

test('ProductPagination', () => {
  const component = renderer.create(
    <ProductPagination totalPages={Math.floor(testProductList.length / 9)} />,
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
