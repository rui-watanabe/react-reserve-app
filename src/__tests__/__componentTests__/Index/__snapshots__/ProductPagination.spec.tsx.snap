// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductPagination 1`] = `
<div
  className="ui center aligned container"
  style={
    Object {
      "margin": "2em",
    }
  }
>
  <div
    aria-label="Pagination Navigation"
    className="ui pagination menu"
    role="navigation"
  >
    <a
      aria-current={false}
      aria-disabled={false}
      aria-label="First item"
      className="item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="firstItem"
      value={1}
    >
      «
    </a>
    <a
      aria-current={false}
      aria-disabled={false}
      aria-label="Previous item"
      className="item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="prevItem"
      value={1}
    >
      ⟨
    </a>
    <a
      aria-current={true}
      aria-disabled={false}
      className="active item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="pageItem"
      value={1}
    >
      1
    </a>
    <a
      aria-current={false}
      aria-disabled={false}
      className="item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="pageItem"
      value={2}
    >
      2
    </a>
    <a
      aria-current={false}
      aria-disabled={false}
      aria-label="Next item"
      className="item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="nextItem"
      value={2}
    >
      ⟩
    </a>
    <a
      aria-current={false}
      aria-disabled={false}
      aria-label="Last item"
      className="item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="lastItem"
      value={2}
    >
      »
    </a>
  </div>
</div>
`;
